fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java

```diff
@@ -315,11 +316,19 @@ private void shortPutToByte(Object[] column, boolean isNullable, int numRows, lo
     public void copyBatchTinyIntResult(Object columnObj, boolean isNullable, int numRows, long nullMapAddr,
             long columnAddr) {
         Object[] column = (Object[]) columnObj;
-        if (column[0] instanceof BigDecimal) {
+        int firstNotNullIndex = 0;
+        if (isNullable) {
+            firstNotNullIndex = getFirstNotNullObject(column, numRows, nullMapAddr);
+        }
```

Comment:
here, if the column all data is null, so the firstNotNullIndex==numRows, and then check column[firstNotNullIndex], will be out of range?

Reply:
You are right!

Reply:
done